Sources/TensorFlow/Core/LazyTensorOperation.swift

```diff
@@ -272,6 +272,77 @@ class LazyTensorOperation: TensorOperation {
     }
 }
 
+protocol _LazyTensorCompatible {
+    /// The underlying `LazyTensor` (if any).
+    var _lazyTensor: LazyTensor? { get }
+
+    /// Returns `Self` that wraps a concrete `LazyTensor`.
+    /// (Triggers materialization if needed.)
+    var _concreteLazyTensor: Self { get }
+
+    /// Similar to the `concreteLazyTensor` with an additional constraint that
+    /// the underlying concrete `LazyTensor` should be marked to be promoted as
+    /// an input when used in an extracted trace.
+    var _concreteInputLazyTensor: Self { get }
+}
+
+extension _AnyTensorHandle {
+    var _lazyTensor: LazyTensor? {
+        if let handle = self as? LazyTensor {
+            return handle
+        } else {
+            return nil
+        }
+    }
+    var _concreteLazyTensor: LazyTensor { LazyTensor(self._tfeTensorHandle) }
+    var _concreteInputLazyTensor: LazyTensor { LazyTensor(_materialized: self._tfeTensorHandle) }
+}
+
+extension TensorHandle: _LazyTensorCompatible {
+    var _lazyTensor: LazyTensor? { handle._lazyTensor }
+    public var _concreteLazyTensor: TensorHandle { TensorHandle(handle: handle._concreteLazyTensor) }
```

Comment:
The protocol is internal. Why does this requirement need to be public? Same for other occurrences of `public`.